### PR TITLE
[CI] Modify CI settings

### PR DESCRIPTION
**Pull request by yl-1993**
Refer to
- [[CI] Cancel previous runs that are not completed](https://github.com/open-mmlab/mmcv/pull/1537)
- [Skip CI when some specific files were changed](https://github.com/open-mmlab/mmcv/pull/1533)
- [Use the official markdownlint hook](https://github.com/open-mmlab/mmcv/pull/1535)
- Add [codespell](https://github.com/codespell-project/codespell) hook

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -7,30 +7,24 @@ on:
   push:
    branches:
     - main
+   paths-ignore:
+     - 'README.md'
+     - 'README_zh-CN.md'
```
Comment (LazyBusyYang) [inline .github/workflows/build.yml:12]: Chinese readme file in root directory is `README_CN.md`.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -7,30 +7,24 @@ on:
...
+     - 'docs/**'
+     - 'docs_zh-CN/**'
 
   pull_request:
+   paths-ignore:
+     - 'README.md'
+     - 'README_zh-CN.md'
```
Comment (LazyBusyYang) [inline .github/workflows/build.yml:19]: Same as above

**Comment by LazyBusyYang** [comment]
We have not hooked markdownlint in pre-commit yet. See mmdet for official markdownlint:
https://github.com/open-mmlab/mmdetection/blob/master/.pre-commit-config.yaml#L31
Note: markdownlint in mmpose uses a non-official repo.

**Comment by yl-1993** [comment]
Previously, we do not include markdownlint hook as it causes some trouble for developers, mainly due to the ruby and gem version on clusters.
This hook is added now and potential problem can be resolved by the following commands:
```
# install rvm
\curl -sSL https://get.rvm.io | bash
# install ruby >= 2.5.1p57, RubyGems >= 2.7.6
rvm install "ruby-2.7.1
```

**Comment by caizhongang** [comment]
Can you update the link to Wishlist in README.md to this link please?
https://github.com/open-mmlab/mmhuman3d/discussions/47

File: mmhuman3d/core/visualization/visualize_smpl.py [hunk 3]
```diff
@@ -616,7 +616,7 @@ def render_smpl(
             will be in the same color.
 
             3). If you want to define your specific color, use an `array`
-            of shape (3,) for singe person and (N, 3) for multiple person.
+            of shape (3,) for single person and (N, 3) for multiple person.
```
Comment (caizhongang) [inline mmhuman3d/core/visualization/visualize_smpl.py:619]: multiple `persons`.

File: mmhuman3d/core/visualization/renderer/torch3d_renderer/base_renderer.py [hunk 4]
```diff
@@ -204,7 +204,7 @@ def init_renderer(self, cameras, lights):
                 dict(type=self.shader_type)))
         return renderer
 
-    def write_images(self, rgbs, valid_masks, images, indexs):
+    def write_images(self, rgbs, valid_masks, images, indexes):
```
Comment (caizhongang) [inline mmhuman3d/core/visualization/renderer/torch3d_renderer/base_renderer.py:207]: I think a more common plural form of `index` is `indices` than `indexes`.

Comment (yl-1993) [reply mmhuman3d/core/visualization/renderer/torch3d_renderer/base_renderer.py:207]: Yes, codespell recommends both `indices` and `indexes`. For other mm repos, it seems that they use [`indices`](https://github.com/open-mmlab/mmdetection/search?q=indices) and [`indexes`](https://github.com/open-mmlab/mmdetection/search?q=indexes) interchangable.

**Comment by yl-1993** [comment]
> Can you update the link to Wishlist in README.md and README_CN.md to this [link](https://github.com/open-mmlab/mmhuman3d/discussions/47) please?

Link of Wishlist has been updated.

File: mmhuman3d/core/visualization/renderer/torch3d_renderer/base_renderer.py (hunk 4 above)
Comment (yl-1993) [reply mmhuman3d/core/visualization/renderer/torch3d_renderer/base_renderer.py:207]: As discussed offline, we may address this issue in future PR, after unifying the convention of ind/idx/index/indices/indexes.

**Comment by LazyBusyYang** [comment]
Seems good. Merging to main.

**Review by LazyBusyYang — APPROVED** [review]

**Pull request merged**